index.bs

```diff
@@ -1300,253 +1300,359 @@ Promise-Using Specifications</a></strong> guideline.
 
 <h3 id="always-add-event-handlers">Always add event handler attributes</h3>
 
-For an object that inherits from {{EventTarget}}, there are two techniques available for registering
-an event handler (e.g., an event named "somethingchanged"):
+If your API adds a new event type,
+add a corresponding <code>on<em>yourevent</em></code> [=event handler IDL attribute=]
+to the interface of any {{EventHandler}} which may handle the new event.
 
-1. {{EventTarget/addEventListener()}} which allows authors to register for the event using the
-    event's name (i.e.,
-    <code highlight="js">someobject.addEventListener("somethingchanged", myhandler)</code>) and
-2. `onsomethingchanged` IDL attributes which allow one event handler to be directly assigned to the
-    object (i.e., <code highlight="js">someobject.onsomethingchanged</code>).
-
-Because there are two techniques for registering events on objects inheriting from {{EventTarget}},
-authors may be tempted to omit the corresponding [=event handler IDL attributes=]. They may assume
-that event handler IDL attributes are a legacy registration technique or are simply not needed
-given that {{EventTarget/addEventListener()}} is available as an alternative. However, it is
-important to continue to define event handler IDL attributes because:
+it's important to continue to define event handler IDL attributes because:
 
 * they preserve consistency in the platform
 * they enable feature-detection for the supported events (see [[#feature-detect]])
 
-So, if the object inherits from {{EventTarget}}, add a corresponding
-<code>on<em>yourevent</em></code> [=event handler IDL attribute=] to the interface.
-
-<p class="note">Note that for HTML and SVG elements, it is traditional to add the
-[=event handler IDL attributes=] on the {{GlobalEventHandlers}} interface mixin, instead of
-directly on the relevant element interface(s). Similarly, [=event handler IDL attributes=]
-are traditionally added to {{WindowEventHandlers}} rather than {{Window}}.</p>
+<p class="note">
+For consistency, if the event needs to be handled by HTML and SVG elements,
+add the [=event handler IDL attributes=]
+on the {{GlobalEventHandlers}} interface mixin,
+instead of directly on the relevant element interface(s).
+Similarly, add [=event handler IDL attributes=]
+to {{WindowEventHandlers}} rather than {{Window}}.
+</p>
 
 
 <h3 id="events-are-for-notification">Events are for notification</h3>
 
-Try to design DOM events to deliver after-the-fact notifications of changes. It may be tempting to try to trigger side-effects from the action of {{EventTarget/dispatchEvent()}}, but in general this is <a href="https://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0510.html">strongly discouraged</a> as it requires changes to the DOM specification when added. Your design will proceed more quickly if you avoid this pattern.
+Issue: I can't understand this one well enough to rewrite it.
```

Comment:
I think it should definitely be re-written without the mailing list reference. How about “DOM events should deliver after-the-fact notifications of changes. They should not be designed to trigger additional actions.” Agree we need some examples.

Reply:
So an example for being "after-the-fact" is that when a `resize` event fires, the size of the element should be (as reported by `getBoundingClientRect`, etc.) the new size rather than the old size.  (And I think if multiple elements are changing size at the same time, probably all of their sizes should be the new sizes rather than the old sizes by the time the first `resize` event fires.)

I think an example for triggering additional actions would be that the script manually creating and dispatching a `resize` event shouldn't actually cause anything to resize.

Reply:
I attempted a rewrite of this in the latest change. PTAL?